lib/puppet_https.rb

```diff
@@ -76,10 +79,27 @@ def post(url, request_body=nil)
   def delete(url)
     url = URI.parse(url)
 
-    request = Net::HTTP::Delete.new(url.request_uri)
+    request = Net::HTTP::Delete.new(url.request_uri, self.auth_header)
     request.content_type = 'application/json'
 
     res = make_ssl_request(url, request)
     res
   end
+
+  #private
+
+  def token
+    return @token if @token
+    token_path = @settings['token_path']
```

Comment:
Do you think it makes sense to have a sensible default token path to read in? I don't think the path is configurable in PE, so I think it'll always be at `HOMEDIR/.puppetlabs/token`...I think?